README.md

```diff
@@ -109,11 +78,70 @@ s3://mybucket/aws-instance-scheduler/v1.4.0/instance-scheduler.template
 s3://mybucket/aws-instance-scheduler/v1.4.0/instance-scheduler-remote.template
 ```
 
-## Deploy
+### Deploy
 
 See the [AWS Instance Scheduler Implementation Guide](https://s3.amazonaws.com/solutions-reference/aws-instance-scheduler/latest/instance-scheduler.pdf) for deployment instructions, using the link to the instance-scheduler.template from your bucket, rather than the one for AWS Solutions. Ex. https://mybucket.s3.amazonaws.com/aws-instance-scheduler/v1.4.0.mybuild/instance-scheduler.template
 
-## CDK Documentation
+
+# Testing the Solution
+
+## Running Tests Locally
+to test the solution you will need to install tox
+
+```pip install tox```
+
+Then from the root directory of the solution
+```
+//run all unit tests
+tox
+
+//test just the lambda code
+tox -e lambda
+
+//test just the cdk code
+tox -e cdk
+```
+
+## Automated Testing Pipeline
+AWS Instance Scheduler comes with an automated testing pipeline that can be used to automatically test any changes you
+develop for the solution on your own development fork
+
+#### _Prerequisites - You must have an AWS account and a fork of this repo_
+
+### Step 1 - Connect CodeStar to Your Github Account
+
+For the pipeline to be able to test your changes, you must provide permission for CodeStar to 
+download the code from your development repo
+
+https://docs.aws.amazon.com/dtconsole/latest/userguide/connections-create-github.html
+
+_note: codestar only needs access to your Instance Scheduler development fork, it does not need access to all repositories_
+
+Once the connection has been setup, make sure you save the connection ARN for the next step
+
+### Step 2 -  Setup Pipeline Parameters
+Go to [Systems Manager Parameter Store](https://us-east-1.console.aws.amazon.com/systems-manager/parameters) 
+and configure the following string parameters
+
+- /InstanceScheduler-build/connection/arn    -- the CodeStar connection ARN from the previous step
+- /InstanceScheduler-build/connection/owner  -- the github owner of your fork
+- /InstanceScheduler-build/connection/repo   -- the repo name of your fork
+- /InstanceScheduler-build/connection/branch -- The branch in your fork that you want to test
```

Comment:
Maybe give examples? Don't give examples for a real repo or fork, or people will set them up and wonder why it's not working.

Reply:
hmm, good point, but I think the only good example I could give for this would be to use the repo itself as an example:

owner: aws-solutions
repo: aws-instance-scheduler
branch: develop